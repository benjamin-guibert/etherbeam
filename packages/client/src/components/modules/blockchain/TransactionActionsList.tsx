import React, { ReactElement, useState } from 'react'
import PaginatedList from 'components/common/PaginatedList'
import Table from 'react-bootstrap/Table'
import { TransactionAction } from 'libraries/ethereum/types'
import TransactionActionRow from './TransactionActionRow'
import { Pagination } from 'libraries/api'

const ACTIONS_PER_PAGE = 20

interface TransactionActionsListProps {
  actions: TransactionAction[]
}

const TransactionActionsList = ({ actions }: TransactionActionsListProps): ReactElement => {
  const [page, setPage] = useState<number>(1)
  const pagination: Pagination = {
    currentPage: page,
    itemsPerPage: ACTIONS_PER_PAGE,
    totalItems: actions.length,
    totalPages: Math.ceil(actions.length / ACTIONS_PER_PAGE),
  }
  const { currentPage, itemsPerPage } = pagination
  const currentPageActions = actions.slice((currentPage - 1) * itemsPerPage, currentPage * itemsPerPage)

  return (
    <>
      <h2>Actions</h2>
      <PaginatedList pagination={pagination} setPage={setPage}>
        <Table striped variant="dark" className="shadow-lg" size="sm">
          <thead>
            <tr className="text-center">
              <th></th>
              <th>Action</th>
              <th>Time</th>
              <th className="d-none d-sm-table-cell">Holder</th>
              <th></th>
            </tr>
          </thead>
          <tbody>
            {currentPageActions.map((action, index) => (
              <TransactionActionRow key={index} action={action} />
            ))}
          </tbody>
        </Table>
      </PaginatedList>
    </>
  )
}

export default TransactionActionsList
