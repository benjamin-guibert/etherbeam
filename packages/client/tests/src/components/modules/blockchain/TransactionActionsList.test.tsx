import * as React from 'react'
import { mount, ReactWrapper } from 'enzyme'
import TransactionActionsList from 'components/modules/blockchain/TransactionActionsList'

describe('<TransactionActionsList />', () => {
  let component: ReactWrapper

  beforeAll(() => {
    component = mount(<TransactionActionsList actions={[]} />)
  })

  it('should render', () => expect(component.find('TransactionActionRow')).toHaveLength(0))

  it('should match snapshot', () => expect(component.render()).toMatchSnapshot())
})
