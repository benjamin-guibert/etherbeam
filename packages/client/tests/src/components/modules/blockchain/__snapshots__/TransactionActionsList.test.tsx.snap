// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TransactionActionsList /> should match snapshot 1`] = `
Array [
  <h2>
    Actions
  </h2>,
  <table
    class="shadow-lg table table-dark table-sm table-striped"
  >
    <thead>
      <tr
        class="text-center"
      >
        <th />
        <th>
          Action
        </th>
        <th>
          Time
        </th>
        <th
          class="d-none d-sm-table-cell"
        >
          Holder
        </th>
        <th />
      </tr>
    </thead>
    <tbody />
  </table>,
]
`;
